src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Frameworks/SupportedTargetFrameworksProvider.cs

```diff
@@ -0,0 +1,105 @@
+            ConfiguredProject project,
+            IProjectSubscriptionService subscriptionService)
+            : base(project, synchronousDisposal: true, registerDataSource: false)
+        {
+            _subscriptionService = subscriptionService;
+        }
+
+        [ConfiguredProjectAutoLoad]
+        [AppliesTo(ProjectCapability.DotNet)]
+        public void Load()
+        {
+            // To avoid UI delays when opening the AppDesigner for the first time, 
+            // we auto-load so that we are included in the first design-time build
+            // for the project.
+            EnsureInitialized();
+        }
+
+        protected override IDisposable? LinkExternalInput(ITargetBlock<EnumCollectionProjectValue> targetBlock)
+        {
+            IProjectValueDataSource<IProjectSubscriptionUpdate> source = _subscriptionService.ProjectBuildRuleSource;
+
+            // Transform the changes from design-time build -> Supported target frameworks
+            DisposableValue<ISourceBlock<EnumCollectionProjectValue>> transformBlock = source.SourceBlock.TransformWithNoDelta(
+                update => update.Derive(Transform),
+                suppressVersionOnlyUpdates: true,
+                ruleNames: SupportedTargetFrameworkAlias.SchemaName);
+
+            // Set the link up so that we publish changes to target block
+            transformBlock.Value.LinkTo(targetBlock, DataflowOption.PropagateCompletion);
+
+            // Join the source blocks, so if they need to switch to UI thread to complete 
+            // and someone is blocked on us on the same thread, the call proceeds
+            JoinUpstreamDataSources(source);
+
+            return transformBlock;
+        }
+
+        private static EnumCollection Transform(IProjectSubscriptionUpdate input)
+        {
+            IProjectRuleSnapshot snapshot = input.CurrentState[SupportedTargetFrameworkAlias.SchemaName];
+
+            return snapshot.Items.Select(ToEnumValue)
+                                    .ToList();
+        }
+
+        private static IEnumValue ToEnumValue(KeyValuePair<string, IImmutableDictionary<string, string>> item)
+        {
+            return new PageEnumValue(new EnumValue()
+            {
+                // Example: <SupportedTargetFrameworkAlias  Include="net5.0-windows"
+                //                                          DisplayName=".NET 5.0" />
+
+                Name = item.Key,
+                DisplayName = item.Value[SupportedTargetFrameworkAlias.DisplayNameProperty]
+            });
+        }
+
+        public Task<IDynamicEnumValuesGenerator> GetProviderAsync(IList<NameValuePair>? options)
+        {
+            return Task.FromResult<IDynamicEnumValuesGenerator>(this);
+        }
+
+        public async Task<EnumCollection> GetListedValuesAsync()
+        {
+            using (JoinableCollection.Join())
+            {
+                EnumCollectionProjectValue snapshot = await SourceBlock.ReceiveAsync();
```

Comment:
When this change was briefly merged into master, I saw tasks getting stuck here with `await SourceBlock.ReceiveAsync()` never completing. Do you have any idea why this might not complete, and do we need to do anything further to guarantee that it does?

Reply:
This will block if the configuration is not active or one of the implicit active configurations because they are not enabled to build. It's impossible to produce data for that, either we block, throw, or return nothing.